### ISSUE TITLE
Fish likes to auto-suggest emacs auto-save files

### ISSUE THREAD

**Issue by laslowh** [fish-shell/fish-shell#985]
This may not be a bug per-se, but fish sure does like to try to get me to open emacs autosave files (those ending in '~').

Try this:
- mkdir /tmp/test
- cd /tmp/test
- touch foo.c foo.c~
- # start typing ' emacs foo.c'

Notice that the suggestion is 'foo.c~', which is almost never the file I want to open.

I say it may not be a bug, because it may be easy for end-users to configure this problem away.  But it seems like a poor out-of-the-box experience for emacs users, and is something I believe should be fixed in the default configuration.

**Comment by faho** [comment]
>even though I never want nvim foo to suggest nvim foo.swp, I do want to be able to autocomplete rm foo.swp.

Yup. Just like you'd sometimes want ".DS_Store" (mainly to remove the bugger), ".git" (to poke around git's internals)....

And in the specific case of backup-files, the trailing "~" is removed easily enough (just press backspace twice after accepting the suggestion).

And might I suggest doing something like I do in my init.el:

```elisp
(setq backup-dir (expand-file-name "backup" user-cache-directory))
(setq backup-by-copying-when-linked t)
(make-directory backup-dir t)
(setq backup-directory-alist `((".*" . ,backup-dir)))
(setq autosave-dir (expand-file-name "save" user-cache-directory))
(setq auto-save-list-file-prefix autosave-dir)
(make-directory autosave-dir t)
(setq auto-save-file-name-transforms `((".*" ,autosave-dir t)))
(setq undo-dir (expand-file-name "undo" user-cache-directory))
(make-directory undo-dir t)
(setq undo-tree-history-directory-alist `(("." . ,undo-dir)))
(setq undo-tree-auto-save-history t)
```

(where `user-cache-directory` has been set to conform to $XDG_CACHE_HOME)

------

Long story short: I don't think this is worth it, and nobody has stepped up in five years.

**Comment by floam** [comment]
> even though I never want nvim foo to suggest nvim foo.swp, I do want to be able to autocomplete rm foo.swp.

I think we're conflating autosuggestion and autocompletion here. Excluding files ending in ~ would not have affected your ability to autocomplete `rm foo.c~`. It just wouldn't autosuggest it. It's good for us to think of heuristics to try to make autosuggestions more likely to be useful and exclude less likely to be desired commands. I think this should have been left open.

**Comment by zanchey** [comment]
@faho has "penalised" these files in autosuggestions, so they'll only be suggested if there is really absolutely nothing better, which I think is a sensible approach to try.
